pkg/ast/ast.go

```diff
@@ -935,6 +936,18 @@ type CaseElse struct {
 	Expr Expr
 }
 
+// IfExpr is IF expression node.
+//
+//     IF ( {{.Expr | sql}}, {{.TrueResult | sql}}, {{.FalseResult | sql}} )
+type IfExpr struct {
```

Comment:
Why is this new struct necessary? Could we reuse `CallExpr`?

Reply:
OK, I got your point. I said the `IfExpr` is different from `CallExpr`, coz if it's a expression not a function.